example/src/Main.purs

```diff
@@ -0,0 +1,327 @@
+
+  renderDateTime ∷ State -> Int -> String -> Either String DateTime -> Array (HTML m)
+  renderDateTime s = renderExample dateTimeConfig s.dateTimes
+
+  eval ∷ Query ~> DSL m
+  eval (Set payload next) = do
+    map mustBeMounted $ case payload of
+      SetTime     idx val -> H.query' timeConfig.cp     idx $ H.request $ left <<< Base <<< (SetValue $ map Right val)
+      SetDate     idx val -> H.query' dateConfig.cp     idx $ H.request $ left <<< Base <<< (SetValue $ map Right val)
+      SetDateTime idx val -> H.query' dateTimeConfig.cp idx $ H.request $ left <<< Base <<< (SetValue $ map Right val)
+      SetDuration idx val -> H.query' durationConfig.cp idx $ H.request $ left <<< Base <<< (SetValue $ map Right val)
+      SetInterval idx val -> do
+        res <- H.query' intervalConfig.cp idx $ H.request $ left <<< Base <<< (SetValue $ map Right val)
+        pure $ void res -- <#> (\error ->  D.traceAny {message:"can't update interval", error})
+    pure next
+  eval (HandleMessage payload next) = do
+    case payload of
+      MsgTime     idx (NotifyChange val) -> H.modify \s -> s {times     = insert idx (show val) s.times}
+      MsgDate     idx (NotifyChange val) -> H.modify \s -> s {dates     = insert idx (show val) s.dates}
+      MsgDateTime idx (NotifyChange val) -> H.modify \s -> s {dateTimes = insert idx (show val) s.dateTimes}
+      MsgDuration idx (NotifyChange val) -> H.modify \s -> s {durations = insert idx (show val) s.durations}
+      MsgInterval idx (NotifyChange val) -> H.modify \s -> s {intervals = insert idx (show val) s.intervals}
+    pure next
+
+
+
+type ExampleConfig fmtInput input fmt query out m =
+  { mkFormat :: fmtInput -> Either String fmt
+  , unformat :: fmt -> String -> Either String input
+  , picker :: fmt -> H.Component HH.HTML query Unit out m
+  , handler :: ∀z. Int -> out -> z -> Query z
+  , setter :: ∀z. Int -> Maybe input -> z -> Query z
+  , cp :: CP.ChildPath query ChildQuery Int Slot
+  }
+
+renderExample :: ∀ fmtInput input fmt query out m
+  . ExampleConfig fmtInput input fmt query out m
+  -> Map Int String
+  -> Int
+  -> fmtInput
+  -> Either String input
+  -> Array (HTML m)
+renderExample c items idx fmt' value'= unEither $ do
+  fmt <- c.mkFormat fmt'
+  value <- either (c.unformat fmt) Right value'
+  let cmp = c.picker fmt
+  pure
+    [ HH.slot' c.cp idx cmp unit (HE.input (c.handler idx))
+    , HH.button [ HE.onClick $ HE.input_ $ c.setter idx (Just value)] [ HH.text "reset"]
+    , HH.button [ HE.onClick $ HE.input_ $ c.setter idx Nothing] [ HH.text "clear"]
+    , case lookup idx items of
+        Nothing -> HH.div_ [HH.text "no value is set"]
+        Just val -> HH.div_ [HH.text $ "value: " <> val]
+    ]
+  where
+  unEither :: Either String (Array (HTML m)) -> Array (HTML m)
+  unEither = either (HH.text >>> pure >>> HH.div_ >>> pure) id
+
+timeConfig :: ∀ m. ExampleConfig String Time TimeF.Format Time.Query Time.Message m
```

Comment:
6 type params are a bit hard to visually parse. Maybe newlines or type synonims could help.